session/pingpong/invoice_tracker_test.go

```diff
@@ -109,6 +109,101 @@ func Test_InvoiceTracker_Start_Stop(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func Test_InvoiceTracker_Start_RefusesUnregisteredUser(t *testing.T) {
+	dir, err := ioutil.TempDir("", "invoice_tracker_test")
+	assert.Nil(t, err)
+	defer os.RemoveAll(dir)
+
+	ks := keystore.NewKeyStore(dir, keystore.LightScryptN, keystore.LightScryptP)
+	acc, err := ks.NewAccount("")
+	assert.Nil(t, err)
+
+	mockSender := &MockPeerInvoiceSender{
+		chanToWriteTo: make(chan crypto.Invoice, 10),
+	}
+
+	exchangeMessageChan := make(chan crypto.ExchangeMessage)
+	bolt, err := boltdb.NewStorage(dir)
+	assert.Nil(t, err)
+	defer bolt.Close()
+
+	tracker := session.NewTracker(time.Now)
+	invoiceStorage := NewProviderInvoiceStorage(NewInvoiceStorage(bolt))
+	accountantPromiseStorage := NewAccountantPromiseStorage(bolt)
+	deps := InvoiceTrackerDeps{
+		Peer:                       identity.FromAddress("some peer"),
+		PeerInvoiceSender:          mockSender,
+		InvoiceStorage:             invoiceStorage,
+		TimeTracker:                &tracker,
+		ChargePeriod:               time.Nanosecond,
+		ExchangeMessageChan:        exchangeMessageChan,
+		ExchangeMessageWaitTimeout: time.Second,
+		PaymentInfo:                dto.PaymentPerTime{Price: money.NewMoney(10, money.CurrencyMyst), Duration: time.Minute},
+		ProviderID:                 identity.FromAddress(acc.Address.Hex()),
+		AccountantID:               identity.FromAddress(acc.Address.Hex()),
+		AccountantCaller:           &mockAccountantCaller{},
+		AccountantPromiseStorage:   accountantPromiseStorage,
+		BlockchainHelper:           &mockBlockchainHelper{isRegistered: false},
+	}
+	invoiceTracker := NewInvoiceTracker(deps)
+
+	go func() {
```

Comment:
1. What is the purpose of time.Sleep?
2. If you need to stop invoice tracker maybe you can use `defer invoiceTracker.Stop()`?

Reply:
start is blocking, henceforth the stop needs to be invoked before calling start.